### PR TITLE
update cadence 1.0 json specs with Authorization

### PR DESCRIPTION
**Pull request by ianthpun**
https://github.com/onflow/cadence/issues/3244

`Restricted` is already removed, so we just need to change `authorized` to `authorizations`

### CODE REVIEW

File: versioned_docs/version-1.0/json-cadence-spec.md [hunk 1]
```diff
@@ -720,7 +726,17 @@ Initializer types are encoded a list of parameters to the initializer.
 ```json
 {
   "kind": "Reference",
-  "authorized": true,
+  "authorization": {
```
Comment (ianthpun) [inline versioned_docs/version-1.0/json-cadence-spec.md:729]: this is the main change

File: versioned_docs/version-1.0/json-cadence-spec.md [hunk 2]
```diff
@@ -750,22 +766,22 @@ Initializer types are encoded a list of parameters to the initializer.
   "kind": "Restriction",
```
Comment (jribbink) [inline versioned_docs/version-1.0/json-cadence-spec.md:766]: This no longer exists in Cadence 1.0

**Review by jribbink — CHANGES_REQUESTED** [review]
Nice! I think there are a couple missing pieces:

- `InclusiveRange` needs to be added
- `Capability` value needs an `id` instead of `path`
-  Function purity needs to be added

anything else I'm missing @onflow/cadence ?

File: versioned_docs/version-1.0/json-cadence-spec.md
Comment (jribbink) [inline versioned_docs/version-1.0/json-cadence-spec.md:1]: Looks like there's also another `Reference` type that needs to be updated https://github.com/onflow/cadence-lang.org/blob/4fb33d6289f8eef18ebc414b7ee3a34f1def699a/versioned_docs/version-1.0/json-cadence-spec.md?plain=1#L808

File: versioned_docs/version-1.0/json-cadence-spec.md (hunk 2 above)
Comment (ianthpun) [reply versioned_docs/version-1.0/json-cadence-spec.md]: I've updated to say `Intersection`, but after seeing some of the tests, im not confident this is a good example.

However, this "restricted" bug in the doc seems to be carried over from the migration, @dsainati1 did you update this section in relation to your FLIP last?

https://github.com/onflow/flips/pull/85
https://github.com/onflow/cadence/issues/2267#issuecomment-1673272779

File: versioned_docs/version-1.0/json-cadence-spec.md
Comment (ianthpun) [reply versioned_docs/version-1.0/json-cadence-spec.md:1]: I believe this is a bug in the doc, capabilities don't seem to have any relation AFAIK to authorized, lemme dig further

Comment (ianthpun) [reply versioned_docs/version-1.0/json-cadence-spec.md:1]: updataed

File: versioned_docs/version-1.0/json-cadence-spec.md (hunk 2 above)
Comment (dsainati1) [reply versioned_docs/version-1.0/json-cadence-spec.md]: https://cadence-lang.org/docs/1.0/language/intersection-types this should be the new link

File: versioned_docs/version-1.0/json-cadence-spec.md [hunk 3]
```diff
@@ -338,7 +337,44 @@ Composite fields are encoded as a list of name-value pairs in the order in which
...
-      "kind": "Int",
+      "kind": "Int"
+    }
+  }
+}
+```
+
+---
+
+## Inclusive Range Value
```
Comment (jribbink) [inline versioned_docs/version-1.0/json-cadence-spec.md:348]: We'll need to include the `InclusiveRange` type as well

File: versioned_docs/version-1.0/json-cadence-spec.md [hunk 4]
```diff
@@ -678,6 +708,7 @@ Initializer types are encoded a list of parameters to the initializer.
     <parameter at index 1>,
     // ...
   ],
+  "purity: "view",
```
Comment (ianthpun) [inline versioned_docs/version-1.0/json-cadence-spec.md:711]: @jribbink added view

**Review by jribbink — CHANGES_REQUESTED** [review]

File: versioned_docs/version-1.0/json-cadence-spec.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (ianthpun) [reply versioned_docs/version-1.0/json-cadence-spec.md:711]: resolved

File: versioned_docs/version-1.0/json-cadence-spec.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ianthpun) [reply versioned_docs/version-1.0/json-cadence-spec.md:348]: resolved

File: versioned_docs/version-1.0/json-cadence-spec.md [hunk 5]
```diff
@@ -338,7 +337,44 @@ Composite fields are encoded as a list of name-value pairs in the order in which
...
+
+## Inclusive Range Value
+
+```json
+{
+  "type": "InclusiveRange",
+  "value": {
+    "start": <start_value>,
+    "end": <end_value>,
+    "step": <step_type>
```
[the remark this answers is not in the document]
Comment (ianthpun) [reply versioned_docs/version-1.0/json-cadence-spec.md:356]: thanks for catching that!

File: versioned_docs/version-1.0/json-cadence-spec.md [hunk 6]
```diff
@@ -847,22 +878,50 @@ represented by its type ID.
...
+	"value": {
+		"staticType": {
+		"kind": "InclusiveRange",
+		"element": {
+			"kind": "Int"
+		}
+		}
+	}
+	}
+}
```
[the remark this answers is not in the document]
Comment (ianthpun) [reply versioned_docs/version-1.0/json-cadence-spec.md:926]: updated, i think this is what your looking for

File: versioned_docs/version-1.0/json-cadence-spec.md [hunk 7]
```diff
@@ -847,22 +878,43 @@ represented by its type ID.
...
   }
 }
 ```
+
+## Inclusive Range Type
+
+```json
+{
+  "kind": "InclusiveRange",
+  "element":  <integer_value>
```
Comment (jribbink) [inline versioned_docs/version-1.0/json-cadence-spec.md:907]: nit: 
```suggestion
  "element":  <integer_type>
```

**Comment by ianthpun** [comment]
@dsainati1 could you do a final check on the docs and confirm it looks good? Thanks!

**Review by dsainati1 — APPROVED** [review]

**Review by jribbink — APPROVED** [review]

**Review by turbolent — COMMENTED** [review]
Looks good!

There are still a couple of things that need to get fixed.

**Comment by ianthpun** [comment]
@turbolent sorry Bastian i just merged this, i'll create a new PR with your changes

**Pull request merged**